dash/dash.py

```diff
@@ -396,7 +396,7 @@ def __init__(  # pylint: disable=too-many-statements
     ):
         _validate.check_obsolete(obsolete)
 
-        caller_name = get_caller_name(self.__class__.__name__)
+        caller_name = get_caller_name()
```

Comment:
To limit the impact on existing apps, can we skip this if there's already a `name` (in which case we aren't going to use it anyway)?
```suggestion
        caller_name = None if name else get_caller_name()
```